garrysmod/lua/includes/extensions/net.lua

```diff
@@ -69,24 +69,31 @@ end
 --
 -- Read/Write a color to/from the stream
 --
-function net.WriteColor( col )
+function net.WriteColor( col, writeAlpha )
+	if ( writeAlpha == nil ) then writeAlpha = true end
 
 	assert( IsColor( col ), "net.WriteColor: color expected, got ".. type( col ) )
 
-	net.WriteUInt( col.r, 8 )
-	net.WriteUInt( col.g, 8 )
-	net.WriteUInt( col.b, 8 )
-	net.WriteUInt( col.a, 8 )
+	local r, g, b, a = col:Unpack()
+	net.WriteUInt( r, 8 )
+	net.WriteUInt( g, 8 )
+	net.WriteUInt( b, 8 )
 
+	if ( writeAlpha ) then
```

Comment:
Can we do `not wroteAlpha` and remove line 73?

Reply:
Since @robotboy655 doesn't like default args in comments in the header, I like the way the first line looks at implicitly signaling the arg defaults to true. If changing it will get this merged faster/at all, though, I'll do it.

Reply:
Depends on what rubat's opinion is right now

Reply:
> Since @robotboy655 doesn't like default args in comments in the header

I am not sure what you mean by that.

Reply:
> 

``function net.WriteColor( writeAlpha --[[= true]] )``

Reply:
Ah. Yeah not a fan, and its is not done like so anywhere else in the codebase.

I think its fine in its current form, the only alternative I see is changing the arg to `noAlpha`, but I am not sure which once would be more convenient.

Reply:
NoAlpha sounds fine.
Since truth of this value will mean "no alpha". Or "dontWriteAlpha"

Reply:
I prefer the writealpha so that false == dont write and true == do write. Seems more intuitive.

Reply:
Ok then maybe explicitly expect true to be passed?

Reply:
That is what is done currently :) It defaults to true to maintain backwards compatibility/write alpha by default.